plugins/all-modules-page/build.gradle.kts

```diff
@@ -29,4 +39,8 @@ dependencies {
 
     val jsoup_version: String by project
     implementation("org.jsoup:jsoup:$jsoup_version")
-}
\ No newline at end of file
+
+    testImplementation(project(":test-utils"))
+    testImplementation(project(":core:test-api"))
+    testImplementation("org.junit.jupiter:junit-jupiter:5.6.0")
```

Comment:
Can we put these into `org.jetbrains.conventions.kotlin-jvm`? Might include deps  above:

```kotlin
    compileOnly(project(":core"))
    implementation(kotlin("stdlib-jdk8"))
    implementation(kotlin("stdlib"))
    implementation(kotlin("reflect"))
```

Reply:
There's no hard and fast rule for this, but I think it's best not to define dependencies in convention plugins (even test dependencies). 

It's much easier to understand what a subproject requires and the links between subprojects when the dependencies are defined locally, because it's easy to look at a subproject's `build.gradle.kts` to check. If the dependencies are applied somewhere else, then they could be really tricky to trace down and understand.

Either way - first let's to convert the existing project to use convention plugins, and to match the existing setup as closely as possible and to make this PR as straightforward as possible